### DISCUSSION TITLE
Detect whether a Reach API has been called by a specific Reach application

### DISCUSSION THREAD

**Discussion by lorenzopolidori** [reach-sh/reach-lang#1351] [Q&A] [answered]
I am calling a Reach API implemented in a Reach application from a separate Reach application using `remote`. Both applications are deployed on the Algorand consensus network.

Is it possible to detect whether the API in the called contract is called using `remote` from a specific contract? I would like to restrict how the API is called in the called contract, i.e. users should be able to call the API only going through a designated calling contract.

**Comment by jeapostrophe** [comment, marked as answer]
Yea. You can look at "this" and use https://docs.reach.sh/rsh/compute/#rsh_Contract.addressEq
